internal/cloud/realm/app.go

```diff
@@ -0,0 +1,74 @@
+package realm
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+// App represents basic Realm App data
+type App struct {
+	ID          string `json:"_id"`
+	GroupID     string `json:"group_id"`
+	ClientAppID string `json:"client_app_id"`
+	Name        string `json:"name"`
+}
+
+var (
+	errGroupNotFound = errors.New("group could not be found")
+)
+
+var (
+	appsPathPattern = adminAPI + "/groups/%s/apps"
+)
+
+func (c *client) GetAppsForUser() ([]App, error) {
+	profile, err := c.GetAuthProfile()
+	if err != nil {
+		return nil, err
+	}
+	groupIDs := profile.AllGroupIDs()
+	var appArr []App
+	appArr = make([]App, 0)
+	for _, groupID := range groupIDs {
+		apps, err := c.GetApps(groupID)
+		if err != nil {
+			// Request will fail if any GetApps call fails
+			return nil, err
+		}
+		appSet := make(map[string]bool)
```

Comment:
question: why are we making a set of apps based on name here?  Since we're just getting all apps per groupID now, we should probably just be returning all of those apps, regardless if they share a name.  i might be missing something here?

if we can get rid of this second for loop here, then the rest just becomes

```go
appArr = append(appArr, apps...)
```

Reply:
Good point, this set was redundant and the code looks a lot nicer now